iroh-net-report/src/reportgen.rs

```diff
@@ -87,12 +88,12 @@ impl Client {
     pub(super) fn new(
         net_report: net_report::Addr,
         last_report: Option<Arc<Report>>,
-        port_mapper: Option<portmapper::Client>,
+        #[cfg(not(wasm_browser))] port_mapper: Option<portmapper::Client>,
         relay_map: RelayMap,
-        stun_sock4: Option<Arc<UdpSocket>>,
-        stun_sock6: Option<Arc<UdpSocket>>,
-        quic_config: Option<QuicConfig>,
-        dns_resolver: DnsResolver,
+        #[cfg(not(wasm_browser))] stun_sock4: Option<Arc<UdpSocket>>,
+        #[cfg(not(wasm_browser))] stun_sock6: Option<Arc<UdpSocket>>,
+        #[cfg(not(wasm_browser))] quic_config: Option<QuicConfig>,
+        #[cfg(not(wasm_browser))] dns_resolver: DnsResolver,
```

Comment:
this sucks, but this method already is problematic because it takes too many arguments

Reply:
Yeah this whole pr is just in the "getting it working" stage not yet in the "make it nice" one.
I think separating some of these things will be possible.

Reply:
I've taken these and grouped them together in a `SocketState` struct. This helped reduce arguments a bunch.